src/web/resources/demo.js

```diff
@@ -97,11 +99,25 @@ function bottom_up(tree, cb) {
         tree.condition = bottom_up(tree.condition, cb);
         tree.trueExpr = bottom_up(tree.trueExpr, cb);
         tree.falseExpr = bottom_up(tree.falseExpr, cb);
+    } else if (tree.blocks) {
+        for (var i = 0; i < tree.blocks.length - 1; i++) {
+            stmt = tree.blocks[i].node;
+            if (stmt.type != "AssignmentNode")
+                throw SyntaxError("Only assignment statements are supported! " + stmt);
```

Comment:
Are we 100% sure that this is always true? Or should we check it in the `tree_errors` function?

Reply:
I'll double check.